OpenBSD does not have an equivalent of Linux's procfs, so even if we obtain the PID, we cannot determine the process path. Therefore, leave the PID as 0.
Index: mozc/src/ipc/unix_ipc.cc
--- mozc/src/ipc/unix_ipc.cc.orig
+++ mozc/src/ipc/unix_ipc.cc
@@ -28,7 +28,7 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 // __linux__ only. Note that __ANDROID__/__wasm__ don't reach here.
-#if defined(__linux__)
+#if defined(__linux__) || defined(__OpenBSD__)
 
 #include <fcntl.h>
 #include <sys/select.h>
@@ -121,10 +121,10 @@ bool IsWriteTimeout(int socket, absl::Duration timeout
 bool IsPeerValid(int socket, pid_t *pid) {
   *pid = 0;
 
-  struct ucred peer_cred;
-  int peer_cred_len = sizeof(peer_cred);
+  struct sockpeercred peer_cred;
+  socklen_t peer_cred_len = sizeof(peer_cred);
   if (getsockopt(socket, SOL_SOCKET, SO_PEERCRED, &peer_cred,
-                 reinterpret_cast<socklen_t *>(&peer_cred_len)) < 0) {
+              &peer_cred_len) < 0) {
     LOG(ERROR) << "cannot get peer credential. Not a Unix socket?";
     return false;
   }
@@ -134,8 +134,6 @@ bool IsPeerValid(int socket, pid_t *pid) {
     return false;
   }
 
-  *pid = peer_cred.pid;
-
   return true;
 }
 
@@ -250,9 +248,6 @@ void IPCClient::Init(absl::string_view name,
       continue;
     }
     sockaddr_un address = {};
-    const size_t server_address_length =
-        (server_address.size() >= UNIX_PATH_MAX) ? UNIX_PATH_MAX - 1
-                                                 : server_address.size();
     if (server_address.size() >= UNIX_PATH_MAX) {
       LOG(WARNING) << "too long path: " << server_address;
     }
@@ -265,7 +260,8 @@ void IPCClient::Init(absl::string_view name,
     address.sun_family = AF_UNIX;
     absl::SNPrintF(address.sun_path, sizeof(address.sun_path), "%s",
                    server_address);
-    const size_t sun_len = sizeof(address.sun_family) + server_address_length;
+    const socklen_t sun_len = SUN_LEN(&address);
+    address.sun_len = sun_len;
     pid_t pid = 0;
     if (::connect(socket_, reinterpret_cast<const sockaddr *>(&address),
                   sun_len) != 0 ||
@@ -381,7 +377,8 @@ IPCServer::IPCServer(absl::string_view name, int32_t n
   int on = 1;
   ::setsockopt(socket_, SOL_SOCKET, SO_REUSEADDR, reinterpret_cast<char *>(&on),
                sizeof(on));
-  const size_t sun_len = sizeof(addr.sun_family) + server_address_.size();
+  const socklen_t sun_len = SUN_LEN(&addr);
+  addr.sun_len = sun_len;
   if (is_file_socket) {
     // Linux does not use files for IPC.
     ::chmod(server_address_.c_str(), 0600);
